.github/workflows/e2e_test.yaml

```diff
@@ -0,0 +1,191 @@
+          EOF
+
+      - name: Cache github-runner Charm
+        uses: actions/cache@v3
+        id: cache-charm
+        with:
+          path: github-runner_ubuntu-22.04-amd64.charm
+          key: github-runner-charm-${{ hashFiles('**/*') }}
+
+      - name: Setup LXD
+        if: steps.cache-charm.outputs.cache-hit != 'true'
+        uses: canonical/setup-lxd@main
+
+      - name: Install charmcraft
+        if: steps.cache-charm.outputs.cache-hit != 'true'
+        run: sudo snap install charmcraft --classic
+
+      - name: Pack github-runner Charm
+        if: steps.cache-charm.outputs.cache-hit != 'true'
+        run: charmcraft pack || ( cat ~/.local/state/charmcraft/log/* 1 && exit 1 )
+
+      - name: Upload github-runner Charm
+        uses: actions/upload-artifact@v3
+        with:
+          name: dangerous-test-only-github-runner_ubuntu-22.04-amd64.charm
+          path: github-runner_ubuntu-22.04-amd64.charm
+
+  run-id:
+    name: Generate Run ID
+    runs-on: ubuntu-latest
+    outputs:
+      run-id: ${{ steps.run-id.outputs.run-id }}
+    steps:
+      - name: Generate Run ID
+        id: run-id
+        run: |
+          echo "run-id=e2e-$(LC_ALL=C tr -dc 'a-z' < /dev/urandom | head -c8)" >> $GITHUB_OUTPUT
+
+  deploy-e2e-test-runner:
+    name: Deploy End-to-End Test Runner
+    runs-on: ubuntu-latest
+    needs: [ build-charm, run-id ]
+    steps:
+      - name: Setup Lxd Juju Controller
+        uses: charmed-kubernetes/actions-operator@main
+        with:
+          juju-channel: 3.1/stable
+          provider: lxd
+
+      - name: Install GitHub Cli
+        run: which gh || sudo apt install gh -y
+
+      - name: Create Testing Juju Model
+        run: juju add-model testing
+
+      - name: Set Testing Model Proxy Configuration
+        run: |
+          juju model-config juju-http-proxy=$http_proxy
+          juju model-config juju-https-proxy=$https_proxy
+          juju model-config juju-no-proxy=$no_proxy
+
+      - name: Change Testing Model Logging Level
+        run: juju model-config logging-config="<root>=INFO;unit=DEBUG"
+
+      - name: Download github-runner Charm
+        uses: actions/download-artifact@v3
+        with:
+          name: dangerous-test-only-github-runner_ubuntu-22.04-amd64.charm
+
+      - name: Deploy github-runner Charm
+        run: |
+          cp github-runner_ubuntu-22.04-amd64.charm /home/$USER/github-runner_ubuntu-22.04-amd64.charm
+          juju deploy /home/$USER/github-runner_ubuntu-22.04-amd64.charm \
+            ${{ needs.run-id.outputs.run-id }} \
+            --base ubuntu@22.04 \
+            --config path=${{ secrets.E2E_TESTING_REPO }} \
+            --config token=${{ secrets.E2E_TESTING_TOKEN }} \
+            --config virtual-machines=1 \
+            --config test-mode=insecure
+
+      - name: Watch github-runner
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        timeout-minutes: 30
+        run: |
+          juju debug-log --replay --tail &
```

Comment:
I suppose these logs are there for debug purpose ?
The model is configured with level DEBUG, this can produce a lot of logs here

Reply:
Yes, it's intended. The debug messages are very helpful.